virtual_feature.pl

```diff
@@ -1483,24 +1485,28 @@ sub feature_get_web_php_children
 {
 my ($d) = @_;
 my $mode = &feature_get_web_php_mode($d);
+my $childrenmax = 
+	defined(&virtual_server::get_php_max_childred_allowed) ? 
+	&virtual_server::get_php_max_childred_allowed() :
+	$virtual_server::max_php_fcgid_children;
 if ($mode eq "fcgid") {
 	# Stored in the domain's config
-	return $d->{'nginx_php_children'} || 1;
+	return $d->{'nginx_php_children'} || 0;
 	}
 elsif ($mode eq "fpm") {
 	# Read from FPM config file
-        my $conf = &virtual_server::get_php_fpm_config();
-        return -1 if (!$conf);
-        my $file = $conf->{'dir'}."/".$d->{'id'}.".conf";
-        my $lref = &read_file_lines($file, 1);
-        my $childs = 0;
-        foreach my $l (@$lref) {
-                if ($l =~ /pm.max_children\s*=\s*(\d+)/) {
-                        $childs = $1;
-                        }
-                }
-        &unflush_file_lines($file);
-        return $childs == 9999 ? 0 : $childs;
+	my $conf = &virtual_server::get_php_fpm_config();
```

Comment:
Isn't there a proper function in  Virtualmin to read the FPM config file, instead of parsing text here?

Reply:
> Isn't there a proper function in Virtualmin to read the FPM config file, instead of parsing text here?


Jamie, I think this comment of yours should belong to some other change? `my $conf = &virtual_server::get_php_fpm_config();` was not changed. I only fixed indents for that line and lines below. The actual change was made to the line containing `9999`. Have another look:

<img width="659" alt="image" src="https://user-images.githubusercontent.com/4426533/188263961-6609c875-e748-4774-a75e-6ff51eb50027.png">

Reply:
You're right ... commit this, and I'll fix that code separately.

Reply:
See https://github.com/virtualmin/virtualmin-nginx/commit/996cb0e240065be43502449c136b9c77329ee299